core/src/main/java/org/apache/iceberg/avro/AvroSchemaUtil.java

```diff
@@ -219,6 +219,13 @@ public static boolean isKeyValueSchema(Schema schema) {
     return schema.getType() == RECORD && schema.getFields().size() == 2;
   }
 
+  public static boolean isVariantSchema(Schema schema) {
+    return schema.getType() == RECORD
+        && schema.getFields().size() == 2
+        && schema.getField("metadata") != null
+        && schema.getField("value") != null;
```

Comment:
If this is checking the metadata/value fields, should this also verify that they are binary?